src/Hooks.ts

```diff
@@ -16,9 +16,14 @@ export function useNavigation<S>(): NavigationScreenProp<S & NavigationRoute> {
 }
 
 export function useNavigationParam<T extends keyof NavigationParams>(
-  paramName: T
+  paramName: T,
+  fallbackValue?
 ) {
-  return useNavigation().getParam(paramName);
+  const { getParam, setParams } = useNavigation()
+  return [
+    getParam(paramName, fallbackValue),
+    (newValue: NonNullable<NavigationParams[T]>) => setParams({ [paramName]: newValue }),
```

Comment:
Maybe worth wrapping this in `React.useCallback` to preserve the reference?

Reply:
Which part exactly?

Reply:
The function which is used to update the param

Reply:
I don't know if it's that much of a performance difference to just memoize the reference to that function.

Reply:
It's not about performance optimisation due to memoizing, but it will avoid extra step to memoize it if you are using `React.memo` or `React.PureComponent`. iirc, the updater function from `React.useState` preserves the reference for this reason